src/Illuminate/Validation/Rules/RegExp.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Illuminate\Validation\Rules;
+
+use Illuminate\Contracts\Support\Arrayable;
+use Illuminate\Support\Str;
+use Illuminate\Support\Stringable as SupportStringable;
+use Stringable;
+
+class RegExp implements Stringable
+{
+    protected string $regExp;
+    protected bool $negated = false;
+
+    /**
+     * @var string[]
+     */
+    protected array $flags = [];
+
+    /**
+     * Create a new array rule instance.
+     *
+     * @param  string  $regExp
+     * @param  string[]|\Illuminate\Contracts\Support\Arrayable|null  $flags
+     * @return void
+     */
+    public function __construct(string $regExp, $flags = null)
+    {
+        $this->regExp = $regExp;
+        $this->flags(...$flags);
+    }
+
+    public function active(bool $active)
+    {
+        $this->active = $active;
+
+        return $this;
+    }
+
+    /**
+     * @param  string[]|\Illuminate\Contracts\Support\Arrayable|null  $flags
+     */
+    public function flags($flags = null)
+    {
+        $this->flags = match (true) {
+            $flags instanceof Arrayable => $flags->toArray(),
+            ! is_array($flags) => func_get_args(),
+            default => $flags,
+        };
+
+        return $this;
+    }
+
+    public function flag(string $flag)
+    {
+        $this->flags = array_unique([...$this->flags, $flag]);
+
+        return $this;
+    }
+
+    public function not()
+    {
+        $this->negated = true;
+
+        return $this;
+    }
+
+    /**
+     * Convert the rule to a validation string.
+     *
+     * @return string
+     */
+    public function __toString()
+    {
+        return (string) Str::of($this->negated ? 'not_regex' : 'regex')
+            ->append(
+                ':',
+                $this->regExp,
+            )
+            ->when($this->flags, function (SupportStringable $str) {
```

Comment:
What if we get all of the flags from the expression when we create the instance? This way if we call `->flags('i', 'g', ...)` (which current replaces *all* flags with the flags provided), it will not keep any of the flags originally in the expression. Then we can just add `implode('', $this->flags)` to the `->append(...)` here.

Reply:
Maybe we could do something like this for the constructor:

```php
public function __construct(string $expression, array|Arrayable $extraFlags = [])
{
    $currentFlags = str_split(Str::of($expression)->afterLast('/'));

    if ($extraFlags instanceof Arrayable) {
        $extraFlags = $extraFlags->toArray();
    }
    
    $this->expression = Str::of($expression)->beforeLast('/')->append('/')->toString();
    $this->flags = array_unique([...$currentFlags, ...$extraFlags]);
}
```

Reply:
That's what I assumed! Now with the updated constructor code, it makes much more sense to me. Good work!

Reply:
> Good work!

Primarily due to your feedback 👍🏻 Thanks a lot!